source/fe/fe_hermite.cc

```diff
@@ -0,0 +1,840 @@
+
+    unsigned int offset = 0;
+
+    AssertDimension(h2l.size(), dim_dofs_2d * dim_dofs_2d);
+
+    // Assign DOFs at nodes
+    for (unsigned int di = 0; di < 2; ++di)
+      for (unsigned int dj = 0; dj < 2; ++dj)
+        for (unsigned int dk = 0; dk < 2; ++dk)
+          for (unsigned int dl = 0; dl < 2; ++dl)
+            {
+              for (unsigned int i = 0; i < node_dofs_1d; ++i)
+                for (unsigned int j = 0; j < node_dofs_1d; ++j)
+                  for (unsigned int k = 0; k < node_dofs_1d; ++k)
+                    for (unsigned int l = 0; l < node_dofs_1d; ++l)
+                      h2l[l + k * node_dofs_1d + j * node_dofs_2d +
+                          i * node_dofs_3d + offset] =
+                        l + k * dim_dofs_1d + j * dim_dofs_2d +
+                        i * dim_dofs_3d +
+                        node_dofs_1d * (dl + dk * dim_dofs_1d +
+                                        dj * dim_dofs_2d + di * dim_dofs_3d);
+
+              offset += node_dofs_1d * node_dofs_3d;
+            }
+  }
+
+
+
+  template <int dim>
+  inline std::vector<unsigned int>
+  hermite_hierarchic_to_lexicographic_numbering(const unsigned int regularity)
+  {
+    const std::vector<unsigned int> dpo =
+      get_hermite_dpo_vector(dim, regularity);
+    const dealii::FiniteElementData<dim> face_data(dpo, 1, 2 * regularity + 1);
+    std::vector<unsigned int>            renumbering(face_data.dofs_per_cell);
+
+    hermite_hierarchic_to_lexicographic_numbering<dim>(regularity, renumbering);
+
+    return renumbering;
+  }
+
+
+
+  template <int dim>
+  std::vector<unsigned int>
+  hermite_lexicographic_to_hierarchic_numbering(const unsigned int regularity)
+  {
+    return Utilities::invert_permutation(
+      hermite_hierarchic_to_lexicographic_numbering<dim>(regularity));
+  }
+
+
+
+  template <int dim>
+  inline std::vector<unsigned int>
+  hermite_face_lexicographic_to_hierarchic_numbering(
+    const unsigned int regularity)
+  {
+    return (dim > 1) ? hermite_lexicographic_to_hierarchic_numbering<dim - 1>(
+                         regularity) :
+                       std::vector<unsigned int>();
+  }
+
+
+
+  template <int dim>
+  TensorProductPolynomials<dim>
+  get_hermite_polynomials(const unsigned int regularity)
+  {
+    TensorProductPolynomials<dim> poly_space(
+      Polynomials::HermitePoly::generate_complete_basis(regularity));
+
+    std::vector<unsigned int> renumber =
+      internal::hermite_hierarchic_to_lexicographic_numbering<dim>(regularity);
+    poly_space.set_numbering(renumber);
+
+    return poly_space;
+  }
+
+
+
+  static inline unsigned int
+  binomial(const unsigned int n, const unsigned int i)
+  {
+    unsigned int C = 1, k = 1;
+
+    for (unsigned int j = n; j > i; --j)
+      {
+        C *= j;
+        C /= k++;
+      }
+
+    return C;
+  }
+
+
+
+  template <int xdim, int xspacedim = xdim, typename xNumber = double>
+  class Rescaler

```

Comment:
I would move this into an anonymous namespace. Also, I would remove the reference to this class in the documentation of `FEHermtie`, since the it is implementation detail and users do not have to care that the work is outsourced to this class. Furthermore, could you add some documentation to this class even though it is internal.